File: ios/Classes/UpdraftSdkFlutterPlugin.swift
import Flutter
import UIKit
import Updraft

public class UpdraftSdkFlutterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "updraft_sdk_flutter", binaryMessenger: registrar.messenger())
    let instance = UpdraftSdkFlutterPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initNativeSdk":
      initNativeSdk(call, result: result)
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
          guard let arguments = call.arguments as? [String: Any], !arguments.isEmpty else {
              result(FlutterError(code: "101", message: "Arguments is null or empty", details: nil))
              return
          }

          guard let sdkKey = arguments["sdkKey"] as? String else {
              result(FlutterError(code: "102", message: "sdkKey can not be empty. Please check your UpdraftSettings. Current settings is \(arguments)", details: nil))
              return
          }

          guard let appKey = arguments["appKey"] as? String else {
              result(FlutterError(code: "102", message: "appKey can not be empty. Please check your UpdraftSettings. Current settings is \(arguments)", details: nil))
              return
          }

          Updraft.shared.start(sdkKey: sdkKey, appKey: appKey)

          let baseUrl = arguments["baseUrl"] as? String
          if (baseUrl != nil) {
              Updraft.shared.baseUrl = baseUrl
          }

          print("success")

          result("")
      }
}

File: android/src/main/kotlin/com/appswithlove/updraft/updraft_sdk_flutter/UpdraftSdkFlutterPlugin.kt
package com.appswithlove.updraft.updraft_sdk_flutter

import android.app.Application
import com.appswithlove.updraft.Updraft
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** UpdraftSdkFlutterPlugin */
class UpdraftSdkFlutterPlugin : FlutterPlugin, MethodCallHandler {
    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    private var flutterPluginBinding: FlutterPlugin.FlutterPluginBinding? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        this.flutterPluginBinding = flutterPluginBinding
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "updraft_sdk_flutter")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method == "initNativeSdk") {
            initNativeSdk(call, result)
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        this.flutterPluginBinding = null
        channel.setMethodCallHandler(null)
    }

    private fun initNativeSdk(call: MethodCall, result: Result) {
        val application = flutterPluginBinding?.applicationContext as? Application
        if (application == null) {
            result.error("100", "Wrong initialisation state", null)
            return
        }

        val args = call.arguments() as Map<String, Any>? ?: mapOf()
        if (args.isEmpty()) {
            result.error("101", "Arguments is null or empty", null)
            return
        }

        try {
            val settings = SettingsExtensions.fromMap(args)
            Updraft.initialize(application, settings)
            Updraft.getInstance()?.start()
            result.success("")
        } catch (e: Exception) {
            result.error("102", e.message, e)
        }
    }
}